src/blas/backends/mklcpu/mklcpu_batch.cxx

```diff
@@ -19,399 +19,1999 @@
 
 // Buffer APIs
 
-void gemm_batch(cl::sycl::queue &queue, transpose transa, transpose transb, int64_t m, int64_t n,
-                int64_t k, float alpha, cl::sycl::buffer<float, 1> &a, int64_t lda,
-                int64_t stride_a, cl::sycl::buffer<float, 1> &b, int64_t ldb, int64_t stride_b,
-                float beta, cl::sycl::buffer<float, 1> &c, int64_t ldc, int64_t stride_c,
+void copy_batch(cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<float, 1> &x, int64_t incx,
+                int64_t stridex, cl::sycl::buffer<float, 1> &y, int64_t incy, int64_t stridey,
                 int64_t batch_size) {
     queue.submit([&](cl::sycl::handler &cgh) {
-        auto a_acc = a.get_access<cl::sycl::access::mode::read>(cgh);
-        auto b_acc = b.get_access<cl::sycl::access::mode::read>(cgh);
-        auto c_acc = c.get_access<cl::sycl::access::mode::read_write>(cgh);
+        auto accessor_x = x.template get_access<cl::sycl::access::mode::read>(cgh);
+        auto accessor_y = y.template get_access<cl::sycl::access::mode::read_write>(cgh);
+        host_task<class mkl_kernel_scopy_batch_strided>(cgh, [=]() {
+            int64_t i;
+            for (i = 0; i < batch_size; i++) {
+                ::cblas_scopy(n, accessor_x.get_pointer() + i * stridex, incx,
```

Comment:
That could potentially work, I guess. Do you know what it was implemented this way in oneMKL?

Reply:
There is no cblas_?copy_batch in oneMKL so we cannot call it directly when defining the DPC++ copy_batch API.
Here you have access to the DPC++ copy_batch API directly so you do not need to reimplement what is done in oneMKL.

Reply:
That would create inconsistency though (and additional dependency on mkl_sycl library?) For all CPU implementations (L1, L2, L3...) we are calling cblas_* interfaces from oneMKL.

Reply:
Yes i was thinking this should probably be done for all functions :-) 
So for CPU backend there is some requirement to not have mkl_sycl library dependencies?

Reply:
I am not aware of such requirement to not have mkl_sycl library dependency for mklcpu backend. @mkrainiuk do you know? (I was just pointing out the additional dependency :) )

Overall, I agree with your proposal. It will simply the code. Since it is impacting all APIs, I think we should do this separately. :) We can bundle this with using DPC++ external APIs for mklgpu backend as well.

Reply:
Initially we enabled mklcpu blas backend without dependencies on mkl_sycl library because opensource compiler runtime and Intel DPC++runtime were not aligned, now it's better, but mkl_sycl library still doesn't fully support opensource compiler runtime, so you can see LAPACK domain failures because it uses mkl_sycl in mklcpu.

Reply:
Thanks @mkrainiuk looks like we are not ready to switch.